File: takka_mobile/lib/core/orders/order_readiness.dart
/// Menu item order readiness — keep labels in sync with web `order-readiness.ts`.
class OrderReadinessOption {
  const OrderReadinessOption({
    required this.id,
    required this.label,
    required this.hint,
  });

  final String id;
  final String label;
  final String hint;
}

const orderReadinessFieldLabel = 'جاهزية الطلب';
const orderReadinessCustomerQuestion = 'متى يكون جاهز؟';

const List<OrderReadinessOption> orderReadinessOptions = [
  OrderReadinessOption(
    id: 'AVAILABLE_NOW',
    label: 'متاح الآن',
    hint: 'استلام أو توصيل في أقرب وقت حسب المطبخ',
  ),
  OrderReadinessOption(
    id: 'PREORDER',
    label: 'حجز مسبق',
    hint: 'يحتاج تجهيز مسبق (مثل التسليم في اليوم التالي)',
  ),
  OrderReadinessOption(
    id: 'ASK_KITCHEN',
    label: 'اسأل المطبخ',
    hint: 'تواصل مع المطبخ لمعرفة إمكانية وموعد الاستلام أو التوصيل',
  ),
];

String orderReadinessLabel(String? value) {
  for (final option in orderReadinessOptions) {
    if (option.id == value) {
      return option.label;
    }
  }
  return orderReadinessOptions.first.label;
}

bool isOrderReadinessId(String value) {
  return orderReadinessOptions.any((option) => option.id == value);
}

String normalizeOrderReadiness(String? value) {
  if (value != null && isOrderReadinessId(value)) {
    return value;
  }
  return orderReadinessOptions.first.id;
}

File: admin/src/lib/order-readiness.ts
/** Order readiness for menu items — keep in sync with Flutter `order_readiness.dart`. */

export const ORDER_READINESS_VALUES = [
  "AVAILABLE_NOW",
  "PREORDER",
  "ASK_KITCHEN",
] as const;

export type OrderReadinessValue = (typeof ORDER_READINESS_VALUES)[number];

export type OrderReadinessOption = {
  id: OrderReadinessValue;
  label: string;
  hint: string;
};

export const ORDER_READINESS_FIELD_LABEL = "جاهزية الطلب";
export const ORDER_READINESS_CUSTOMER_QUESTION = "متى يكون جاهز؟";

export const ORDER_READINESS_OPTIONS: OrderReadinessOption[] = [
  {
    id: "AVAILABLE_NOW",
    label: "متاح الآن",
    hint: "استلام أو توصيل في أقرب وقت حسب المطبخ",
  },
  {
    id: "PREORDER",
    label: "حجز مسبق",
    hint: "يحتاج تجهيز مسبق (مثل التسليم في اليوم التالي)",
  },
  {
    id: "ASK_KITCHEN",
    label: "اسأل المطبخ",
    hint: "تواصل مع المطبخ لمعرفة إمكانية وموعد الاستلام أو التوصيل",
  },
];

export function isOrderReadinessValue(
  value: string,
): value is OrderReadinessValue {
  return ORDER_READINESS_VALUES.includes(value as OrderReadinessValue);
}

export function getOrderReadinessLabel(value: string | null | undefined): string {
  const match = ORDER_READINESS_OPTIONS.find((option) => option.id === value);
  return match?.label ?? ORDER_READINESS_OPTIONS[0].label;
}

export function parseOrderReadiness(
  value: string | null | undefined,
): OrderReadinessValue {
  if (value && isOrderReadinessValue(value)) {
    return value;
  }
  return "AVAILABLE_NOW";
}
